src/util/StoreProperty.js

```diff
@@ -0,0 +1,40 @@
+export class StoreProperty {
+	_observer = new Set();
+	_state = {};
+
+	constructor(value) {
+		this._state = { ...value, ...this.properties() };
+
+		Object.entries(this._state).map(([key, value]) => {
+			Object.defineProperty(this, key, {
+				get: () => {
+					return this._state[key];
+				},
+				set: (newValue) => {
+					this._state[key] = newValue;
+					this.notifyObserver();
+				},
+			});
+		});
+	}
+
+	properties() {
+		return {};
+	}
+
+	valueOf() {
+		return this._state;
+	}
+
+	toString() {
+		return `${this._state}`;
+	}
+
+	registerObserver(observer) {
```

Comment:
@quarkus Is it `register` or rather `add`? Isn't it more like adding an event listener? adding it to the Set also uses `add`

Reply:
Yes but value.add looks super akward in the BaseElement. I changed the naming to observe / unobserve to make it more explicit.

Reply:
Yes I can imagine that!